gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/structure/io/IoTest.java

```diff
@@ -249,6 +250,63 @@ public void shouldProperlyEncodeWithGraphML() throws Exception {
             // need to manually close the "g2" instance
             graphProvider.clear(g2, configuration);
         }
+
+        @Test
+        @FeatureRequirement(featureClass = Graph.Features.EdgeFeatures.class, feature = Graph.Features.EdgeFeatures.FEATURE_ADD_EDGES)
+        @FeatureRequirement(featureClass = EdgePropertyFeatures.class, feature = FEATURE_STRING_VALUES)
+        @FeatureRequirement(featureClass = Graph.Features.VertexFeatures.class, feature = Graph.Features.VertexFeatures.FEATURE_ADD_VERTICES)
+        public void shouldReadWriteSelfLoopingEdges() throws Exception {
+            final Configuration sourceConf = graphProvider.newGraphConfiguration("source", this.getClass(), name.getMethodName(), null);
```

Comment:
@laxatives you don't need to create a new graph here as you get one for free by virtue of starting a test.  Just assign the "graph" variable to your "source" variable on the line below.  please do that for the other tests as well.